sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/trees/SQLQueryContextSuite.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.trees
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+
+class SQLQueryContextSuite extends SparkFunSuite with SQLHelper {
+
+  test("SPARK-40136: the length of the fragment in query context") {
```

Comment:
this basically tests `String.substring`... can we have an end-to-end test?

Reply:
@cloud-fan The failed test `DIVIDE_BY_ZERO: can't divide an integer by zero *** FAILED ***` is an end-to-end test actually. Let me fix it.

Reply:
Great! Can we remove this test suite then? The code we are testing is simply `sqlText.get.substring(originStartIndex.get, originStopIndex.get + 1)`

Reply:
I wouldn't remove this unit tests because:
1. Current implementation of `fragment` is not only `substring()`. It is a combination of isValid + substring.
2. The end-to-end tests don't test corner cases.